src/corehost/cli/test/mockhostpolicy/mockhostpolicy.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <iostream>
+#include "pal.h"
+#include "trace.h"
+#include "error_codes.h"
+#include "host_interface.h"
+
+std::vector<char> tostr(const pal::char_t* value)
+{
+    std::vector<char> vect;
+    pal::pal_utf8string(pal::string_t(value), &vect);
+    return vect;
+}
+
+void print_strarr(const char* prefix, const strarr_t& arr)
+{
+    if (arr.len == 0)
+    {
+        std::cout << prefix << "<empty>" << std::endl;
+        return;
+    }
+
+    for (int i = 0; i < arr.len; i++)
+    {
+        std::cout << prefix << tostr(arr.arr[i]).data() << std::endl;
+    }
+}
+
+SHARED_API int corehost_load(host_interface_t* init)
```

Comment:
I am getting a bit leery of exported APIs that don't have explicit calling conventions. I have been here less than a year and have been impacted by and fixed no fewer than 8 calling convention bugs in various parts of the runtime. We need to be explicit on _all_ exports going forward so when browsing code it is obvious what to expect. @janvorli recently addressed this in [`coreclrhost.h`](https://github.com/dotnet/coreclr/pull/22801) and his good example should be followed.

cc @jkoritzinsky

Reply:
If there's anything more to do, I think it should be done in the `SHARED_API` macro. We use that consistently for all exports. Do you want to "separate" these somehow?

Reply:
We can't put calling conventions in the `SHARED_API` since the convention is required to be between the return type and name - https://docs.microsoft.com/cpp/cpp/calling-example-function-prototype-and-call. I think clang is a bit more reasonable about this.

Reply:
I see - I updated #5135 to add explicit calling convention to all APIs.